sklearn/decomposition/_factor_analysis.py

```diff
@@ -159,6 +161,18 @@ class FactorAnalysis(_ClassNamePrefixFeaturesOutMixin, TransformerMixin, BaseEst
     (1797, 7)
     """
 
+    _parameter_constraints = {
+        "n_components": [Interval(Integral, 0, None, closed="left"), None],
```

Comment:
We need at least 1 component
```suggestion
        "n_components": [Interval(Integral, 1, None, closed="left"), None],
```

Reply:
There is a test case that should pass with n_components=0:
https://github.com/scikit-learn/scikit-learn/blob/405e6fe9172b613148293f0ca72d0945ddca24d5/sklearn/decomposition/tests/test_factor_analysis.py#L81-L83

Reply:
Right my bad. Not sure how 0 component is useful but let's keep it.
tranform doesn't work with 0 components, though